src/arch/MovieTexture/MovieTexture_Generic.cpp

```diff
@@ -33,40 +34,54 @@ MovieTexture_Generic::MovieTexture_Generic( RageTextureID ID, MovieDecoder *pDec
 	m_bLoop = true;
 	m_pSurface = nullptr;
 	m_pTextureLock = nullptr;
-	m_ImageWaiting = FRAME_NONE;
-	m_fRate = 1;
-	m_bWantRewind = false;
+
 	m_fClock = 0;
-	m_bFrameSkipMode = false;
 	m_pSprite = new Sprite;
+	m_fRate = 1;
 }
 
 RString MovieTexture_Generic::Init()
 {
-	RString sError = m_pDecoder->Open( GetID().filename );
+	RString sError = m_pDecoder->Open(GetID().filename);
 	if( sError != "" )
 		return sError;
 
 	CreateTexture();
 	CreateFrameRects();
 
-	/* Decode one frame, to guarantee that the texture is drawn when this function returns. */
-	int ret = m_pDecoder->DecodeFrame( -1 );
-	if( ret == -1 )
-		return ssprintf( "%s: error getting first frame", GetID().filename.c_str() );
-	if( ret == 0 )
-	{
-		/* There's nothing there. */
-		return ssprintf( "%s: EOF getting first frame", GetID().filename.c_str() );
-	}
 
-	m_ImageWaiting = FRAME_DECODED;
+	// Draw the first frame immediately, to guarantee that the texture is drawn
+	// when this function returns--if possible.
+	if (m_pDecoder->DecodeNextFrame() < 0) {
+		LOG->Trace("Failure to decode first frame of video file \"%s\"", GetID().filename.c_str());
+		m_failure = true;
+		return RString("Failure to display movie.");
+	};
+	UpdateFrame();
+	typedef std::chrono::high_resolution_clock Time;
+	typedef std::chrono::milliseconds ms;
+	typedef std::chrono::duration<float> fsec;
+	using std::chrono::operator""ms;
```

Comment:
You might want to use a RageTimer method to retrieve microseconds instead of implementing `<chrono>` here to ensure it's consistent with the same clock the rest of the game is using. If it has to remain float, there is `RageTimer::GetTimeSinceStartFast()`.

Reply:
Thanks for the tip--in this location at least the timer is really only for debugging and diagnostics. Changed.